icu4j/maven-build/pom.xml

```diff
@@ -0,0 +1,244 @@
+    <system>JIRA</system>
+    <url>https://unicode-org.atlassian.net/projects/ICU</url>
+  </issueManagement>
+
+  <distributionManagement>
+    <repository>
+      <id>icu4j-releases</id>
+      <name>ICU4J Central Repository</name>
+      <url>https://oss.sonatype.org/service/local/staging/deploy/maven2</url>
+    </repository>
+    <snapshotRepository>
+      <id>icu4j-snapshots</id>
+      <name>ICU4J Central Development Repository</name>
+      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+    </snapshotRepository>
+  </distributionManagement>
+
+  <properties>
+    <!-- ICU4J currently builds with Java 8 sources -->
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <!-- Keep the target class file versions consistent with the source file version -->
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <!-- The encoding / encoding form of the source files -->
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+  <!-- 
+    Note: the .jar files in ./local-maven-repo are copies of the files at
+    icu4j/main/shared/data/*.jar
+  -->
+  <!--
+    Note: this overall solution of a local filesystem repository is based on the
+    answer here: https://stackoverflow.com/a/28762617/2077918
+    The directory structure and files were created using the `mvn install-file` command, 
+    with the copied .jar files being replaced by symlinks to avoid duplicating files on
+    developers' local filesystems.
+    Ex: for version 73.1-SNAPSHOT, when running the current directory (<ICU>/icu4j/maven-build):
+    ```
+    mvn org.apache.maven.plugins:maven-install-plugin:2.5.2:install-file  \
+    -Dfile=../main/shared/data/icudata.jar \
+    -DgroupId=com.ibm.icu -DartifactId=icudata \
+    -Dversion=73.1-SNAPSHOT -Dpackaging=jar \
+    -DlocalRepositoryPath=./local-maven-repo \
+    -DgeneratePom=true
+
+    mvn org.apache.maven.plugins:maven-install-plugin:2.5.2:install-file  \
+    -Dfile=../main/shared/data/icutzdata.jar \
+    -DgroupId=com.ibm.icu -DartifactId=icutzdata \
+    -Dversion=73.1-SNAPSHOT -Dpackaging=jar \
+    -DlocalRepositoryPath=./local-maven-repo \
+    -DgeneratePom=true
+
+    mvn org.apache.maven.plugins:maven-install-plugin:2.5.2:install-file  \
+    -Dfile=../main/shared/data/testdata.jar \
+    -DgroupId=com.ibm.icu -DartifactId=testdata \
+    -Dversion=73.1-SNAPSHOT -Dpackaging=jar \
+    -DlocalRepositoryPath=./local-maven-repo \
+    -DgeneratePom=true
+    ```    
+  -->
+
+  <modules>
+    <module>maven-icu4j</module>
+    <module>maven-icu4j-datafiles</module>
+    <module>maven-icu4j-charset</module>
+    <module>maven-icu4j-localespi</module>
+    <module>maven-icu4j-test-datafiles</module>
+  </modules>
+
+  <build>
+    <pluginManagement>
+      <plugins>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.8.1</version>
```

Comment:
Personally I like moving the versions in `<properties>`
Then you can easily see them in the same place, update versions, see what's used.

Same for the dependency versions.

There it is even nicer when you have a "family" of plugins with the same version
(imagine someone using icu and wanting the 3 artifacts, they define `icu4j.version` and use it for `icu4j`, `icu4j-charset`, `icu4j-localespi`)

Reply:
Done.  Yeah, I left this for cleanup at the end because of how tricky and time consuming getting the other stuff to work was ("to work" = to match the Ant .jar file contents with no more & no less; to be import-able into IntelliJ and figure out VS Code workarounds; etc.).